tests/pytests/test_stats.py

```diff
@@ -391,3 +391,46 @@ def testInfoIndexingTime(env):
 
     d = index_info(env, 'idx2')
     env.assertGreater(float(d['total_indexing_time']), 0)
+
+def testKeyTableInfo(env):
+    conn = getConnectionByEnv(env)
+    env.cmd('FT.CREATE', 'idx', 'PREFIX', 1, 'b:', 'SCHEMA', 'txt', 'TEXT')
+
+    d = index_info(env, 'idx')
+    key_table_size = float(d['key_table_size_mb'])
+    env.assertEqual(key_table_size, 0)
+
+    # Add and delete keys, size should remain the same as the initial size
+    conn.execute_command('HSET', 'b:1', 'txt', '1')
+    conn.execute_command('HSET', 'b:10', 'txt', '10')
+    conn.execute_command('HSET', 'b:15', 'txt', '15')
+    conn.execute_command('DEL', 'b:10')
+    conn.execute_command('DEL', 'b:1') # Term node with a single child
+    conn.execute_command('DEL', 'b:15')
+    d = index_info(env, 'idx')
+    env.assertEqual(float(d['key_table_size_mb']), key_table_size)
+
+    # Delete a node with children, should not affect the key table size, it
+    # only affects the cardinality of the trie
+    d = index_info(env, 'idx')
+    conn.execute_command('HSET', 'b:1', 'txt', '1')
+    conn.execute_command('HSET', 'b:10', 'txt', '10')
+    conn.execute_command('HSET', 'b:15', 'txt', '15')
+    key_table_size = float(d['key_table_size_mb'])
+    conn.execute_command('DEL', 'b:1')
+    env.assertEqual(float(d['key_table_size_mb']), key_table_size)
+    
+    # update a key, should not affect the key table size
+    conn.execute_command('HSET', 'b:1', 'txt', '1')
+    conn.execute_command('HSET', 'b:10', 'txt', '10')
+    conn.execute_command('HSET', 'b:15', 'txt', '15')
```

Comment:
Are these updates needed?

Reply:
No. They are not needed, I removed them.